parcels/field.py

```diff
@@ -251,6 +252,7 @@ def __init__(
         self.V = V
         self.W = W
         self.grid = U.grid
+        self.igrid = U.igrid
```

Comment:
`igrid` is not being updated in the code - so all the indexing is done on the last grid (perhaps leading to the change in hash that we saw elsewhere?) Should we resolve that in another PR?

Changes required:
```
FieldSet.__init__
   - make the field addition use `add_field` instead of plainly constructing the 
     dict of fields
FieldSet.add_field()
   - update fieldset.fields
   - update fieldset.gridset (also make this a normal variable instead of a property)
   - update field.igrid indexing the grid position in the updated gridset

```

Reply:
Ah, I didn't realise that `grid` still required some plumbing. Now created #2221 to track progress